src/mutation.cpp

```diff
@@ -695,10 +695,14 @@ void player::mutate_towards(std::string mut)
         }
     }
 
-    if (!cancel.empty()) {
-        std::string removed = cancel[ rng(0, cancel.size() - 1) ];
-        remove_mutation(removed);
-        return;
+    for (size_t i = 0; i < cancel.size(); i++) {
+        if (!cancel.empty()) {
+            std::string removed = cancel[i];
+            remove_mutation(removed);
+            cancel.erase(cancel.begin() + i);
+            i--;
+            return;
```

Comment:
It probably isn't, though as I recall, removing it caused separate problems.  Will take another look tomorrow or Saturday.